### DISCUSSION TITLE
Past regressor using RecursiveTabularRegressionForecaster with LGBM

### DISCUSSION THREAD

**Discussion by Miky14880** [alan-turing-institute/sktime#2515] [Q&A]
Hello!
first of all, let me thank for the marvelous package, great job!
I am actually using RecursiveTabularRegressionForecaster within TransformedTargetForecaster, LightGBM as estimator and I tried to add a past covariate.
At predict step I have anyway to pass X with same length as fh, is it ok to put as value np.nan since I only have past data?

Any suggestion is welcome!
Thanks in advance, regards
Michela

**Comment by fkiraly** [comment]
@Miky14880, this would work only if (a) your pipeline regressor can deal with NA and/or exogeneous data only in the past, or (b) your setup with the pipeline is such that the nan get thrown away, e.g., if you have a lag of `len(fh)` or higher, say, in a `WindowSummarizer`.

Generally, I would advise to try and see whether it breaks... if it breaks and you feel it shouldn´t , I´d encourage you to report it here. The pipeline functionality is not fully consolidated, and every bug report helps.
